### PR TITLE
Exposing the `indexQueryMetrics` to the `CosmosConfig` via the `application.properties` configuration file

### PR DESCRIPTION
**Pull request by trande4884**
# Description

Exposing the `indexQueryMetrics` to the `CosmosConfig` via the `application.properties` configuration file.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Thanks @trande4884 the changes look good to me, however, can we please add some more tests for repositories use cases (other than CosmosTemplateIT and ReactiveCosmosTemplateIT)

**Comment by trande4884** [comment]
> Thanks @trande4884 the changes look good to me, however, can we please add some more tests for repositories use cases (other than CosmosTemplateIT and ReactiveCosmosTemplateIT)

Done

**Review by kushagraThapar — APPROVED** [review]
Thanks @trande4884, looks good to me.

File: sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/config/CosmosConfig.java [hunk 1]
```diff
@@ -118,6 +123,35 @@ public CosmosConfig(ResponseDiagnosticsProcessor responseDiagnosticsProcessor,
...
+     *
+     * @param responseDiagnosticsProcessor must not be {@literal null}
+     * @param databaseThroughputConfig may be {@literal null}
+     * @param queryMetricsEnabled must not be {@literal null}
+     * @param indexMetricsEnabled must not be {@literal null}
+     * @param maxDegreeOfParallelism must not be {@literal null}
+     * @param maxBufferedItemCount must not be {@literal null}
+     * @param responseContinuationTokenLimitInKb must not be {@literal null}
+     */
+    @ConstructorProperties({"responseDiagnosticsProcessor", "databaseThroughputConfig", "queryMetricsEnabled", "maxDegreeOfParallelism", "maxBufferedItemCount", "responseContinuationTokenLimitInKb"})
```
Comment (xinlian12) [inline sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/config/CosmosConfig.java:143]: missing `indexMetricsEnabled`?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Review by kushagraThapar — APPROVED** [review]
Thanks for adding further tests @trande4884 , they are great!

**Pull request merged**